packages/flutter_web_plugins/pubspec.yaml

```diff
@@ -4,7 +4,7 @@ author: Flutter Authors <flutter-dev@googlegroups.com>
 homepage: http://flutter.dev
 
 environment:
-  sdk: ">=2.0.0-dev.28.0 <3.0.0"
+  sdk: ">=2.11.0 <3.0.0"
```

Comment:
If you change the constraints to https://github.com/flutter/flutter/blob/master/packages/flutter/pubspec.yaml#L8 you should not need the experiment (though you may need to restart the IDE)

Reply:
Ahhh I remember trying `2.12.0` and having the compiler complain about my version of dart not being recent enough (but I didn't do the `-0`!) 

I'll try that again.

Reply:
This *does work*, but it's uncovered a warning that I hadn't seen before:

```
The package resolution is not fully migrated to null-safety.            
                                                                        
package:test_api is not opted into null safety in its pubspec.yaml:     
  |                                                                     
8 |   sdk: '>=2.10.0-0 <2.12.0'                                         
  |        ^^^^^^^^^^^^^^^^^^^^                                         
  |                                                                     
                                                                        
Either downgrade your sdk constraint, or invoke dart/flutter with       
`--no-sound-null-safety`.                                               
                                                                        
To learn more about available versions of your dependencies try running 
`pub outdated --mode=null-safety`.                                      
                                                                        
See more at https://dart.dev/null-safety/migration-guide.               
```

Reply:
I have this (autogenerated) dependency on test_api:

```
  test_api: 0.2.19-nullsafety.4 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
```

Reply:
It is a bug in pub, I believe it can be ignored

Reply:
> **0.2.19-nullsafety.5**: Update SDK constraints to `>=2.12.0-0 <3.0.0` based on beta release guidelines.

Reply:
I tried to update-packages, but test_api is stuck in .4?

<details>

<summary><tt>$ flutter update-packages --force-upgrade</tt></summary>

```
Building flutter tool...
Upgrading packages...
WARNING: the following packages use hard-coded version constraints:
  - flutter_gallery_assets: ^0.2.0
  - mockito: 4.1.1
  - vm_service_client: 0.2.6+2
  - flutter_template_images: 1.0.1
  - shelf: 0.7.5
  - async: 2.5.0-nullsafety.2
Skipping boolean_selector because it was not transitive
  - characters: 1.1.0-nullsafety.4
Skipping charcode because it was not transitive
  - clock: 1.1.0-nullsafety.2
  - collection: 1.15.0-nullsafety.4
  - fake_async: 1.2.0-nullsafety.2
  - js: 0.6.3-nullsafety.2
Skipping matcher because it was not transitive
  - meta: 1.3.0-nullsafety.5
  - path: 1.8.0-nullsafety.2
Skipping pedantic because it was not transitive
  - pool: 1.5.0-nullsafety.2
Skipping source_maps because it was not transitive
Skipping source_map_stack_trace because it was not transitive
Skipping source_span because it was not transitive
  - stack_trace: 1.10.0-nullsafety.5
  - stream_channel: 2.1.0-nullsafety.2
  - string_scanner: 1.1.0-nullsafety.2
Skipping term_glyph because it was not transitive
  - test: 1.16.0-nullsafety.7
  - test_api: 0.2.19-nullsafety.4
  - test_core: 0.3.12-nullsafety.7
  - typed_data: 1.3.0-nullsafety.4
  - vector_math: 2.1.0-nullsafety.4
  - platform: 3.0.0-nullsafety.3
  - file: 6.0.0-nullsafety.3
  - process: 4.0.0-nullsafety.3
  - process_runner: 4.0.0-nullsafety.4
  - build_runner_core: 5.2.0
  - build_modules: 2.10.1
  - path_provider: 1.6.14
  - video_player: 2.0.0-nullsafety.1
  - url_launcher: 6.0.0-nullsafety
  - connectivity: 3.0.0-nullsafety
  - device_info: 2.0.0-nullsafety
Running "flutter pub upgrade" in flutter_update_packages.CDJCJS...      2,540ms
Running "flutter pub get" in snippets...                           413ms
Running "flutter pub get" in manual_tests...                       313ms
Running "flutter pub get" in gen_keycodes...                       252ms
Running "flutter pub get" in vitool...                             364ms
Running "flutter pub get" in tools...                              518ms
Running "flutter pub get" in devicelab...                          493ms
Running "flutter pub get" in web...                                306ms
Running "flutter pub get" in platform_interaction...               495ms
Running "flutter pub get" in android_views...                    1,791ms
Running "flutter pub get" in flutter_driver_screenshot_test...      1,297ms
Running "flutter pub get" in android_semantics_testing...        1,291ms
Running "flutter pub get" in flutter_gallery...                  1,470ms
Running "flutter pub get" in external_ui...                      1,248ms
Running "flutter pub get" in gradle_deprecated_settings...         625ms
Running "flutter pub get" in channels...                         1,317ms
Running "flutter pub get" in flutterapp...                         650ms
Running "flutter pub get" in image_loading...                    1,242ms
Running "flutter pub get" in flavors...                          1,247ms
Running "flutter pub get" in release_smoke_test...                 910ms
Running "flutter pub get" in flutterapp...                         771ms
Running "flutter pub get" in web_compile_tests...                  589ms
Running "flutter pub get" in abstract_method_smoke_test...         526ms
Running "flutter pub get" in ios_platform_view_tests...          1,166ms
Running "flutter pub get" in hybrid_android_views...             1,522ms
Running "flutter pub get" in splash_screen_load_rotate...          902ms
Running "flutter pub get" in splash_screen_kitchen_sink...       1,328ms
Running "flutter pub get" in splash_screen_trans_rotate...       1,138ms
Running "flutter pub get" in ui...                               1,248ms
Running "flutter pub get" in ios_app_with_extensions...            669ms
Running "flutter pub get" in non_nullable...                       711ms
Running "flutter pub get" in web_e2e_tests...                    1,225ms
Running "flutter pub get" in android_embedding_v2_smoke_test...        762ms
Running "flutter pub get" in tracing_tests...                      716ms
Running "flutter pub get" in stocks...                           1,102ms
Running "flutter pub get" in microbenchmarks...                  1,259ms
Running "flutter pub get" in complex_layout...                   1,314ms
Running "flutter pub get" in macrobenchmarks...                  1,153ms
Running "flutter pub get" in platform_views_layout...            1,178ms
Running "flutter pub get" in platform_views_layout_hybrid_composition...      1,255ms
Running "flutter pub get" in customer_testing...                   705ms
Running "flutter pub get" in automated_tests...                  1,110ms
Running "flutter pub get" in bots...                             1,294ms
Running "flutter pub get" in platform_view...                      526ms
Running "flutter pub get" in hello_world...                      1,187ms
Running "flutter pub get" in platform_channel_swift...           1,441ms
Running "flutter pub get" in platform_channel...                 1,316ms
Running "flutter pub get" in flutter_view...                       492ms
Running "flutter pub get" in layers...                             735ms
Running "flutter pub get" in image_list...                         713ms
Running "flutter pub get" in splash...                             618ms
Running "flutter pub get" in flutter_driver...                   1,122ms
Running "flutter pub get" in fuchsia_remote_debug_protocol...      1,231ms
Running "flutter pub get" in main...                               272ms
Running "flutter pub get" in flutter_tools...                      895ms
Running "flutter pub get" in test...                               699ms
Running "flutter pub get" in test_private...                       507ms
Running "flutter pub get" in flutter...                            896ms
Running "flutter pub get" in _flutter_web_build_script...        1,251ms
Running "flutter pub get" in flutter_web_plugins...                391ms
Running "flutter pub get" in flutter_goldens...                    856ms
Running "flutter pub get" in flutter_localizations...              685ms
Running "flutter pub get" in project_root...                       164ms
Running "flutter pub get" in flutter_test...                       644ms
Running "flutter pub get" in flutter_goldens_client...             476ms
Downloading lcov data for package:flutter...                       508ms

Ran 'pub' 64 times and fetched coverage data in 57.7s.
```

</details>

Reply:
My branch was created from an older version of flutter. I rebased the branch to the latest tip of master, and this went away.

Reply:
(This is done)